src/remote-data.ts

```diff
@@ -163,7 +163,12 @@ export class RemoteInitial<L, A> {
 	 *
 	 * `success(21).fold(foldInitial, foldPending, foldFailure, foldSuccess) will return 22`
 	 */
-	fold<B>(initial: B, pending: B, failure: Function1<L, B>, success: Function1<A, B>): B {
+	fold<B>(
+		initial: B,
+		pending: Function1<Option<RemoteProgress>, B>,
```

Comment:
Sure I can split the changes and provide the breaking change in a separate pull request?

Reply:
Well the problem is actually deeper. If we start processing `pending`s with functions then we should also change `pending` helper which immediately turns out to be the same as `progress`. Smth one should be left.
/cc @sutarmin  @OliverJAsh

This may be a single PR for the next breaking `0.4` version together with #5